reconcile/utils/promotion_state.py

```diff
@@ -47,28 +47,51 @@ def cache_commit_shas_from_s3(self) -> None:
         """
         all_keys = self._state.ls()
         for commit in all_keys:
-            # Format: /promotions/{channel}/{commit-sha}
-            if not commit.startswith("/promotions/"):
+            # for backwards compatibility - remove this after a while
+            if commit.startswith("/promotions/"):
+                # Format: /promotions/{channel}/{commit-sha}
+                _, _, channel_name, commit_sha = commit.split("/")
+                self._commits_by_channel[channel_name].add(commit_sha)
+            # / for backwards compatibility - remove this after a while
+
+            # Format: /deployments/{saas-target-uid}/{channel}/{commit-sha}
```

Comment:
nit: I wonder if it would make sense to put the channel first:

```
/deployments/{channel}/{saas-target-uid}/{commit-sha}
```

That way, it is easier to navigate through the bucket when debugging sth, because usually you are looking for the channel first.